source/security/index.md

```diff
@@ -289,3 +289,38 @@ If you were using `excludeParams` previously please compare patterns used by yo
 This mechanism was introduced in version 2.5. It allows control what methods can be accessed with the bang "!" operator 
 via [Dynamic Method Invocation](../core-developers/action-configuration.html#dynamic-method-invocation). Please read 
 more in the Strict Method Invocation section of [Action Configuration](../core-developers/action-configuration.html).
+
+### Content Security Policy
+
+[Content Security Policy](https://csp.withgoogle.com/docs/index.html) is a mitigation against XSS and CSS injection attacks. The recommended way to use CSP is to implement strict CSP.
+
+CSP is enabled by setting the `Content-Security-Policy` or `Content-Security-Policy-Report-Only` HTTP response headers, based on whether it’s in enforcing or reporting mode, respectively. In enforcing mode, any event that violates the policy is not executed and if the policy specifies  a `reportUri` a JSON object with the details of the CSP violation is sent to the specified endpoint. In reporting mode, the event is executed normally and a CSP report is being sent to the defined `reportUri`.
+
+CSP comes in many flavors, but the policy we chose to implement is a nonce-based strict-dynamic CSP, which is the most robust policy. This policy requires the browser to check that all scripts have the valid nonce value (which is included in the response header generated by the server) before running them. This ensures that a malicious script not approved (by attaching the nonce value) by the server will not run, thus protecting against XSS attacks. With each response, a new nonce value is generated, so adopting CSP requires implementing a mechanism that automatically attaches the valid nonce value to all `script, link` blocks. Strict CSP also blocks any inline event handlers (`onclick` etc.) and `javascript:` URIs, so adopting this policy requires refactoring by the developers if these were in use. The default recommended policy used is as follows:
+
+```
+Content-Security-Policy:
+  object-src 'none';
+  script-src 'nonce-{random}' 'strict-dynamic' https: http:;
+  base-uri 'none';
+  report-uri https://your-report-collector.example.com/csp-report
+```
+
+The specified policy includes a `report-uri` directive. This should not be confused with the report-only and enforcing modes of CSP. In both modes, if the `report-uri` is specified, the browser will send a JSON blob containing details about the violation as a POST request to the specified `report-uri`.
+
+```
```

Comment:
let's add a line preceding this code block that says that this is a sample report